uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-fs/src/main/java/org/uberfire/java/nio/fs/file/SimpleFileSystemProvider.java

```diff
@@ -252,117 +236,41 @@ private boolean shouldCreateOrOpenAByteChannel(final Set<? extends OpenOption> o
 
     @Override
     public void createDirectory(final Path dir,
-                                final FileAttribute<?>... attrs)
-            throws UnsupportedOperationException, FileAlreadyExistsException, IOException, SecurityException {
+                                final FileAttribute<?>... attrs) {
         checkNotNull("dir",
                      dir);
         final Path realDir = dir.toAbsolutePath();
-        if (realDir.toFile().exists()) {
-            throw new FileAlreadyExistsException(dir.toString());
-        }
+        checkFileExistsThenThrow(realDir);
         realDir.toFile().mkdirs();
     }
 
     @Override
-    public DirectoryStream<Path> newDirectoryStream(final Path dir,
-                                                    final DirectoryStream.Filter<Path> filter)
-            throws NotDirectoryException, IOException, SecurityException {
-        checkNotNull("filter",
-                     filter);
-        final File file = checkNotNull("dir",
-                                       dir).toFile();
+    public DirectoryStream<Path> newDirectoryStream(final Path dir, final DirectoryStream.Filter<Path> filter) {
```

Comment:
why not preserve the original throws? this was more for `documentation` purposes

Reply:
Same, SonarCloud rule violation.

Reply:
ignore sonarCloud :)

Reply:
Reverted back to original style, but SonarCloud might not be happy :(